### DISCUSSION TITLE
Failed to build module 'Quick'; this SDK is not supported by the compiler - Xcode 13

### DISCUSSION THREAD

**Discussion by luchi2300** [Quick/Quick#1109] [Q&A] [answered]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Quick/blob/main/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?
Update the project I am working on to Xcode 13 version. I am using Carthage.

### What did you expect to happen?

Compile without a problem.

### What actually happened instead?

Getting this error: 

Failed to build module 'Quick'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.3.2 (swiftlang-1200.0.45 clang-1200.0.32.28)', while this compiler is 'Apple Swift version 5.5 (swiftlang-1300.0.31.1 clang-1300.0.29.1)'). Please select a toolchain which matches the SDK.

### Environment

List the software versions you're using:

 - Quick: v4.0.0
 - Nimble: "v9.2.1"
 - Xcode Version: Xcode 13.0
 - Swift Version:  Swift 5.5

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Carthage: 0.38.0

**Comment by bclausdorff** [comment]
Generally, you have to recompile your dependencies with Carthage after changing your Swift version (which most often changes when updating Xcode).

So if you haven't already, you can do this via:

`carthage bootstrap`

**Comment by luchi2300** [comment]
Hello, thanks for the reply. Yes, I did everything you mentioned but still the same issue. I tried removing the derived data, Carthage caches, Carthage checkout folder, and still the same issue.

**Comment by bclausdorff** [comment, marked as answer]
Did you make sure, you were using Xcode 13.0 for compiling the dependencies with Carthage? If you got multiple Xcode versions installed on your machine, maybe a different one got used for building.

You can check which Xcode installation is active and will be used by Carthage via

`xcode-select -print-path`

You can switch the active Xcode installation, which will be used by Carthage by passing a path to `xcode-select`. On my machine this looks something like this:

`sudo xcode-select -s /Applications/Xcode-13.0.0.app/Contents/Developer`

For managing multiple Xcode installations on one machine, I can highly recommend [Xcodes](https://github.com/RobotsAndPencils/XcodesApp).

**Comment by luchi2300** [comment]
thanks @bclausdorff the problem was as you mentioned, it was using a different Xcode version for compiling the dependencies, I set the Xcode version at the beginning of the process and it worked perfectly.
